pkg/cmd/server/api/validation/master.go

```diff
@@ -0,0 +1,169 @@
+				errs.NewFieldInvalid("oauthConfig.assetPublicURL", config.OAuthConfig.AssetPublicURL, "must match assetConfig.publicURL"),
+			)
+		}
+
+		// TODO warn when the CORS list does not include the assetConfig.publicURL host:port
+		// only warn cause they could handle CORS headers themselves in a proxy
+	}
+
+	if config.DNSConfig != nil {
+		allErrs = append(allErrs, ValidateHostPort(config.DNSConfig.BindAddress, "bindAddress").Prefix("dnsConfig")...)
+	}
+
+	allErrs = append(allErrs, ValidateEtcdConnectionInfo(config.EtcdClientInfo).Prefix("etcdClientInfo")...)
+	if config.EtcdConfig != nil {
+		allErrs = append(allErrs, ValidateEtcdConfig(config.EtcdConfig).Prefix("etcdConfig")...)
+	}
+
+	allErrs = append(allErrs, ValidateImageConfig(config.ImageConfig).Prefix("imageConfig")...)
+
+	allErrs = append(allErrs, ValidateKubeletConnectionInfo(config.KubeletClientInfo).Prefix("kubeletClientInfo")...)
+
+	if config.KubernetesMasterConfig != nil {
+		allErrs = append(allErrs, ValidateKubernetesMasterConfig(config.KubernetesMasterConfig).Prefix("kubernetesMasterConfig")...)
+	}
+
+	allErrs = append(allErrs, ValidateKubeConfig(config.MasterClients.DeployerKubeConfig, "deployerKubeConfig").Prefix("masterClients")...)
+	allErrs = append(allErrs, ValidateKubeConfig(config.MasterClients.OpenShiftLoopbackKubeConfig, "openShiftLoopbackKubeConfig").Prefix("masterClients")...)
+	allErrs = append(allErrs, ValidateKubeConfig(config.MasterClients.KubernetesKubeConfig, "kubernetesKubeConfig").Prefix("masterClients")...)
+
+	allErrs = append(allErrs, ValidatePolicyConfig(config.PolicyConfig).Prefix("policyConfig")...)
+
+	allErrs = append(allErrs, ValidateServingInfo(config.ServingInfo).Prefix("servingInfo")...)
+
+	return allErrs
+}
+
+func ValidateAssetConfig(config *api.AssetConfig) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
+
+	allErrs = append(allErrs, ValidateServingInfo(config.ServingInfo).Prefix("servingInfo")...)
+
+	urlObj, urlErrs := ValidateURL(config.PublicURL, "publicURL")
+	if len(urlErrs) > 0 {
+		allErrs = append(allErrs, urlErrs...)
+	}
+	if urlObj != nil {
+		if !strings.HasSuffix(urlObj.Path, "/") {
+			allErrs = append(allErrs, errs.NewFieldInvalid("publicURL", config.PublicURL, "must have a trailing slash in path"))
+		}
+	}
+
+	return allErrs
+}
+
+func ValidateEtcdConnectionInfo(config api.EtcdConnectionInfo) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
+
+	if len(config.URLs) == 0 {
+		allErrs = append(allErrs, errs.NewFieldRequired("urls"))
+	}
+	for i, u := range config.URLs {
+		_, urlErrs := ValidateURL(u, fmt.Sprintf("urls[%d]", i))
+		if len(urlErrs) > 0 {
+			allErrs = append(allErrs, urlErrs...)
+		}
+	}
+
+	if len(config.CA) > 0 {
+		allErrs = append(allErrs, ValidateFile(config.CA, "ca")...)
+	}
+	allErrs = append(allErrs, ValidateCertInfo(config.ClientCert, false)...)
```

Comment:
If the etcd server is using a cert signed by a recognized authority, you don't need a custom CA